client/apollo/react/src/Pagination/ItemPagination/ItemPaginationCommon.tsx

```diff
@@ -0,0 +1,47 @@
+import { type ElementType } from "react";
+import { type PolymorphicComponent } from "../../utilities/types/PolymorphicComponent";
+import { getClassName } from "../../utilities/getClassName";
+
+export const ELLIPSIS = "...";
+
+type ItemPaginationCustomProps = {
+  as?: string | ElementType;
+  page: number | typeof ELLIPSIS;
+  isCurrentPage: boolean;
+};
```

Comment:
Avec ce typage on étend pas les props du composant passé en `as`.
Donc si on passe un `Link` de react-router en `as`, on ne pourra pas passer la prop `to`.
(cela simplifierais aussi la gestion du href et du type plus bas)

Reply:
c'est parce que j'ai voulu partager le type avec la méthode getComponent mais finalement je n'ai pas besoin de cette méthode